compaction/compaction_manager.cc

```diff
@@ -26,6 +26,7 @@
 #include "utils/error_injection.hh"
 #include "utils/UUID_gen.hh"
 #include "db/system_keyspace.hh"
+#include "tombstone_gc-internals.hh"
```

Comment:
how about tombstone_gc_internal.hh?

Reply:
I have a secret notation where "tombstone_gc" is a name from the source and "internal" is a concept so they're joined with a dash. I don't mind changing to an underscore.

Reply:
It's also not a forward declaration.

Reply:
As I said, I'm not insisting, and I don't think the way you split it currently indeed fits the "fwd" paradigm. But the problem with an "internals" header file is that neither nor anyone else will ever know what belongs in which header file and why. It doesn't mean that this split isn't right - but it probably won't remain right for long.

Reply:
It would be nice if you could change to tombstone_gc_internal.hh which the naming is more consistent in my view.

Reply:
@asias I don't love either of these options, but Avi's "internals" does sound right - it's the **internals** of this class.

Reply:
In any case, @xemul already merged this PR :-)